src/Integration.UnitTests/Service/ConnectionInformationTests.cs

```diff
@@ -65,9 +65,18 @@ public void ConnectionInformation_WithoutLoginInformation()
             var testSubject2 = (ConnectionInformation)((ICloneable)testSubject).Clone();
 
             // Verify testSubject2
-            Assert.IsNull(testSubject.Password, "Password wasn't provided");
-            Assert.IsNull(testSubject.UserName, "UserName wasn't provided");
-            Assert.AreEqual(serverUri, testSubject.ServerUri, "ServerUri doesn't match");
+            Assert.IsNull(testSubject2.Password, "Password wasn't provided");
+            Assert.IsNull(testSubject2.UserName, "UserName wasn't provided");
+            Assert.AreEqual(serverUri, testSubject2.ServerUri, "ServerUri doesn't match");
         }
+
+        [TestMethod]
+        public void ConnectionInformation_Ctor_NormalizesServerUri()
+        {
+            // Act
+            var noSlashResult = new ConnectionInformation(new Uri("http://localhost/NoSlash"));
+
+            // Verify
+            Assert.AreEqual("http://localhost/NoSlash/", noSlashResult.ServerUri.ToString(), "Unexpected normalisation of URI without trailing slash");        }
```

Comment:
Why not checking the whole Uri? Also, can we compare the Uri's instead? i.e. is Uri(http://x/) == Uri(http://x) ?